tests/bls_tests.rs

```diff
@@ -0,0 +1,341 @@
+use ethereum_consensus::crypto::{
+    aggregate, aggregate_verify, fast_aggregate_verify, PublicKey, SecretKey, Signature,
+};
+use glob::glob;
+use hex;
+use serde::{Deserialize, Serialize};
+use serde_yaml;
+use std::fmt::Debug;
+use std::fs::File;
+
+fn decode_hex_with_prefix<T: AsRef<[u8]> + Debug>(data: T) -> Vec<u8> {
+    // a small check for empty strings
+    if data.as_ref().len() >= 2 {
+        hex::decode(&data.as_ref()[2..]).expect("is well-formed hex")
+    } else {
+        hex::decode(&data.as_ref()).expect("is well-formed hex")
+    }
```

Comment:
something that may be more tolerant to a wider variety of inputs is to just check if there is a prefix `0x` using e.g. `String::starts_with` https://doc.rust-lang.org/stable/std/string/struct.String.html#method.starts_with

something like:
```rust
let data = data.as_ref();
let data = if data.starts_with("0x") {
    &data[2..]
else {
    data
};
hex::decode(data).expect("...")
```


if the empty string is a special case worth checking specifically, you could check `data.is_empty()`

Reply:
> one more thing i'd consider is moving the data parsing into `serde`
> 
> there are field attributes you can use to specify the desired transformation and that should keep the function bodies smaller and more focused by moving the parsing code elsewhere...
> 
> to give an example, `SigningTestIO` would look like:
> 
> ```rust
> #[derive(Debug, Serialize, Deserialize)]
> struct SigningInput {
>     #[serde(deserialize_with = "try_decode_hex_with_prefix")]
>     privkey: Option<SecretKey>,
>     #[serde(deserialize_with = "decode_hex_with_prefix")]
>     message: Vec<u8>,
> }
> 
> #[derive(Debug, Serialize, Deserialize)]
> struct SigningTestIO {
>     input: SigningInput,
>     #[serde(deserialize_with = "decode_hex_with_prefix")]
>     output: Vec<u8>,
> }
> ```
> 
> where you need to add a "fallible" version of `decode_hex_with_prefix` that would return an `None: Option<SecretKey>` if the key fails to parse.
> 
> you would also need to adapt the function signature of `decode_hex_with_prefix` to match the expected signature, here is an example: https://stackoverflow.com/a/46755370 (under the first header with their fn `from_hex`).
> 
> and this would be adjusted for each test type...

On this point, I'm struggling a little to understand how to make the Deserializer sufficiently generic to generate options for `PublicKey`, `Signature` and `SecretKey`.  In the stackoverflow example, the return type is set to `Result<u64>`. I'm trying something like this:
```
fn decode_hex_with_prefix<T: AsRef<[u8]> + Debug>(data: T) -> Vec<u8> {
    let data = data.as_ref();
    let data= if data.starts_with(b"0x") {
        &data[2..] } else {
            &data
        };
        hex::decode(data).expect("is not well-formed hex")
    }
fn try_decode_hex_with_prefix<T: AsRef<[u8]> + Debug>(data: T) -> Option<Vec<u8>> {
    match SecretKey::from_bytes(&decode_hex_with_prefix(data)) {
        Ok(value) => Some(value),
        Err(_) => None,
    }
}
```
but this will only return Option<SecretKey>. How to generalize this?